server/routes/api/attachments/attachments.ts

```diff
@@ -5,46 +5,32 @@ import { bytesToHumanReadable } from "@shared/utils/files";
 import { AttachmentValidation } from "@shared/validations";
 import { AuthorizationError, ValidationError } from "@server/errors";
 import auth from "@server/middlewares/authentication";
-import {
-  transaction,
-  TransactionContext,
-} from "@server/middlewares/transaction";
+import { transaction } from "@server/middlewares/transaction";
+import validate from "@server/middlewares/validate";
 import { Attachment, Document, Event } from "@server/models";
 import AttachmentHelper from "@server/models/helpers/AttachmentHelper";
 import { authorize } from "@server/policies";
 import { presentAttachment } from "@server/presenters";
-import { ContextWithState } from "@server/types";
+import { APIContext, ContextWithState } from "@server/types";
 import { getPresignedPost, publicS3Endpoint } from "@server/utils/s3";
-import { assertIn, assertPresent, assertUuid } from "@server/validation";
+import { assertIn, assertUuid } from "@server/validation";
+import * as T from "./schema";
 
 const router = new Router();
 
 router.post(
   "attachments.create",
   auth(),
+  validate(T.AttachmentsCreateSchema),
   transaction(),
-  async (ctx: TransactionContext) => {
-    const {
-      name,
-      documentId,
-      contentType = "application/octet-stream",
-      size,
-      // 'public' is now deprecated and can be removed on December 1 2022.
-      public: isPublicDeprecated,
-      preset = isPublicDeprecated
-        ? AttachmentPreset.Avatar
-        : AttachmentPreset.DocumentAttachment,
-    } = ctx.request.body;
+  async (ctx: APIContext<T.AttachmentCreateReq>) => {
+    const { name, documentId, contentType, size, preset } = ctx.input;
     const { user, transaction } = ctx.state;
```

Comment:
Here we lose the type for `ctx.state.transaction` because I had to switch the ctx type to `APIContext`. Do you suggest introducing another type for that?

Reply:
The two middlewares are both newish and haven't been used together yet, need to figure out the pattern to enable typing though as these will be used together more and more.